src/OctopusPuppet.Cmd/Program.cs

```diff
@@ -190,16 +194,17 @@ private static IEnumerable<IComponentVertexDeployer> GetDeployers(string targetE
 
         private static int Deploy(DeployOptions opts)
         {
-            var notifier = GetNotifier(opts.HideDeploymentProgress, opts.Teamcity);
+            var logMessager = new OctopusLogMessager(opts.OctopusUrl);
+            var notifier = GetNotifier(logMessager, opts.HideDeploymentProgress, opts.Teamcity);
 
             var environmentDeployment = LoadEnvironmentDeploy(opts.EnvironmentDeploymentPath);
             var deployers = GetDeployers(opts.TargetEnvironment, true, true, opts.OctopusUrl, opts.OctopusApiKey);
-            return Deploy(notifier, environmentDeployment, opts.MaximumParallelDeployments, deployers);
+            return Deploy(notifier, logMessager, environmentDeployment, opts.MaximumParallelDeployments, deployers);
         }
 
         private static int CommandLineParsingError(IEnumerable<Error> errors)
         {
-            var notifier = GetNotifier(false, false);
+            var notifier = GetNotifier(null, false, false);
```

Comment:
Can we change this so we have GetNotifier() by having parameters with default values?

Reply:
Much cleaner now :>

Reply:
Good! Can we do it for other parameters as well?